sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -573,17 +591,32 @@ def fit(self, X, y, sample_weight=None):
                         X_binned_small_train,
                         y_small_train,
                         sample_weight_small_train,
+                        indices_small_train,
                     ) = self._get_small_trainset(
-                        X_binned_train, y_train, sample_weight_train, self._random_seed
+                        X_binned_train,
+                        y_train,
+                        sample_weight_train,
+                        self._random_seed,
                     )
 
+                    # If the scorer is a predefined string, then we optimize
+                    # the evaluation by using the raw predictions.
+                    if scoring_is_predefined_string:
+                        raw_predictions_small_train = raw_predictions[
+                            indices_small_train
+                        ]
```

Comment:
It would be a change of behaviour, but how would the timing change if we use the whole training set in this case as with `self.scoring == "loss"`?

Reply:
With the [same benchmark](https://gist.github.com/thomasjpfan/5c4aff6d17f807801ddef1a73fe6c53b) and using the whole training set, I get ~12% more runtime for regression and ~76% more runtime for classification compare to this PR.

It's still faster `main`, but as you said, using the whole training set changes behavior.